python/pyspark/sql/functions.py

```diff
@@ -2121,33 +2127,35 @@ def wrapper(*args):
 
         wrapper.func = self.func
         wrapper.returnType = self.returnType
-        wrapper.vectorized = self.vectorized
+        wrapper.pythonUdfType = self.pythonUdfType
 
         return wrapper
 
 
-def _create_udf(f, returnType, vectorized):
+def _create_udf(f, returnType, pythonUdfType):
 
-    def _udf(f, returnType=StringType(), vectorized=vectorized):
-        if vectorized:
+    def _udf(f, returnType=StringType(), pythonUdfType=pythonUdfType):
+        if pythonUdfType == PythonUdfType.PANDAS_UDF \
+           or pythonUdfType == PythonUdfType.PANDAS_GROUPED_UDF:
```

Comment:
shall we add the check that `PANDAS_GROUPED_UDF` can only take one parameter?

Reply:
Yes, I'll add it.